stdlib/LibGit2/test/libgit2.jl

```diff
@@ -1933,7 +1933,7 @@ mktempdir() do dir
                 LibGit2.set!(cfg, "credential.useHttpPath", "true")
                 LibGit2.set!(cfg, "credential.https://github.com.useHttpPath", "false")
 
-                @test !LibGit2.use_http_path(cfg, github_cred)
+                @test_broken !LibGit2.use_http_path(cfg, github_cred)
```

Comment:
I'm not sure why this test fails. No related change appears to be mentioned in the release notes at https://github.com/libgit2/libgit2/releases.

Reply:
You could try modifying the regex as mentioned in this comment: https://github.com/JuliaLang/julia/blob/c3f5414d16ea063032bc0c792c632825fb6aec7f/stdlib/LibGit2/src/gitcredential.jl#L271

Reply:
Unfortunately, using the cased regex makes tests on lines 1929 and 1937 fail (i.e. the result is always `false`). That sounds consistent with comments on that issue, which say that the option is lowercased AFAICT.

Reply:
Actually the problem is that the order in which configuration entries are iterated over has changed: it now respects the order in which entries were defined (https://github.com/libgit2/libgit2/issues/4361). AFAICT the current code isn't completely correct since it will prefer a global setting if it was defined last, even if a domain-specific setting exists. I've pushed a commit which should fix this.

BTW, the new order-preserving behavior of libgit2 0.28 means that `credential_helpers` could probably now allow resetting credential helpers (there's a note about that in the code). But I'll leave that for later.

Reply:
Bump @omus. Do you think my solution is fine?

Reply:
The change looks reasonable to me. I may look into adding more tests for this but I can do that when I look into the state of credential resetting